JetBrains plugins docs (#1943)
gitpod-io/website: gitpod/docs/ides-and-editors/goland.md

</figure>

**Important:** You must restart any started workspaces for your IDE preferences to take effect.

## Install Plugins

Plugins extend the core functionality of {title}:

- Integration with issue trackers, build management servers, and other tools.
- Coding assistance support for various languages and frameworks.
- Shortcut hints, live previews, File Watchers, and so on.

> - Coding exercises that can help you to learn a new programming language.

Reviewer: Thought: 

I like the idea of having this context, but I'm wondering if we should link to JetBrains docs regarding benefits/purpose of plugins and keep our reference docs purely about plugin setup 🤔

Author: yes, let's simplify, link and just discuss a difference in our context 👍

> {title} plugins can either be installed on your **local machine** or in a **remote Gitpod workspace**.

Reviewer: It'd be good if we're able to explain why the difference / what the difference is between the remote and local plugins and their limitations. I've not seen that JetBrains had this documented well already somewhere, but I can take another look.

Author: I think we can duplicate it in both places. Depends there user will navigate, but it does not harm to repeat ourself.

Reviewer: Yeah. In which case I'd suggest just to pull up some short description of the difference from the introductions of each section up here, so the user can see side-by-side the difference. 

Suggestion 🙏 

```suggestion
{title} plugins can either be installed in two ways:

 1. **local machine** - These changes apply to all workspaces that the user connects to. 
 2. **Remote Gitpod workspace** - These changes apply to all users who connect to the workspace only. 
 
 More details on the different approaches are given below. 

```

### Install on your local machine

Local plugins should be installed on your local machine, e.g. contributing shortcuts or themes.
Once installed a plugin is available in each **JetBrains Client** and for **any Gitpod workspace**.

To install a local plugin follow these steps:

- In **JetBrains Client** open the IDE settings and select **Plugins**.
- Find the plugin in the **Marketplace** and click **Install**.

<figure>
<img class="shadow-medium w-full rounded-xl max-w-3xl mt-x-small" alt="Install a local plugin for {title}" src="/images/editors/install-local-jb-plugin.png">
    <figcaption>Install a local plugin for {title}</figcaption>
</figure>
For more information on regular plugin management, refer to <a href="https://www.jetbrains.com/help/go/managing-plugins.html" target="_blank">{title} docs</a>.

### Install in Gitpod workspace

Remote plugins require access to remote filesystem and tools, e.g. contributing support of languages and frameworks. Currently a remote plugin can either be installed for the current Gitpod workspace or for repository.

#### Install for the current workspace

> You can install a remote plugin only for you for the current workspace following these steps:

Reviewer: ```suggestion
You can install a remote plugin only for your current workspace following these steps:
```

> You can install a remote plugin only for you for the current workspace following these steps:

Author: I meant a user current workspace here, not any other.

- In **JetBrains Client** open the IDE settings and select **Plugins On Host**.
- Find the plugin in the **Marketplace** and click **Install**.

<figure>
<img class="shadow-medium w-full rounded-xl max-w-3xl mt-x-small" alt="Install a remote plugin for {title}" src="/images/editors/install-remote-jb-plugin.png">
    <figcaption>Install a remote plugin for {title}</figcaption>
</figure>
For more information on regular plugin management, refer to <a href="https://www.jetbrains.com/help/go/managing-plugins.html" target="_blank">{title} docs</a>.

#### Install for repository

You can share a remote plugin with everybody working on the repository by adding it to [.gitpod.yml](/docs/config-gitpod-file) and pushing to your Git repository.
Each workspace is preconfigured with plugins from this configuration file, here is an example:

```yaml
jetbrains:
  goland:

>     - pluginId

Reviewer 2: I guess it would make it clearer for the reader if we list more then one plugin in this dummy list, just to indicate that it's also possible to install two or more plugins to an IDE through this config.

Otherwise, I think we need to explicitly mention it somewhere in this text block.

Author: ok, will put couple plugins, do you have an idea which can be applied to all products and good?

Reviewer 2: Sure, here are some for general use:

- [Prettier](https://plugins.jetbrains.com/plugin/10456-prettier)
- [BashSupport](https://plugins.jetbrains.com/plugin/4230-bashsupport)
- [IdeaVim](https://plugins.jetbrains.com/plugin/164-ideavim)
- [.​ignore](https://plugins.jetbrains.com/plugin/7495--ignore)

More famous plugins with multi-IDE support can be found in the [Top Downloaded Plugins](https://plugins.jetbrains.com/search?orderBy=downloads&products=androidstudio&products=appcode&products=clion&products=dataspell&products=dbe&products=go&products=idea&products=idea_ce&products=idea_edu&products=mps&products=phpstorm&products=pycharm&products=pycharm_ce&products=pycharm_edu&products=rider&products=ruby&products=webstorm) section.

```

You can find **pluginId** on the [JetBrains Marketplace](https://plugins.jetbrains.com) page:

> - find a page of the required plugid;
> - select the **Versions** tab;
> - and click any version to copy **pluginId** (short name such as `org.rust.lang`) of the plugin you want to install.

Reviewer: ```suggestion
- Find a page of the required plugin
- Select the **Versions** tab
- Click any version to copy the **pluginId** (short name such as `org.rust.lang`) of the plugin you want to install.
```
